django/views/decorators/cache.py

```diff
@@ -1,10 +1,15 @@
+import asyncio
 from functools import wraps
 
 from django.middleware.cache import CacheMiddleware
 from django.utils.cache import add_never_cache_headers, patch_cache_control
-from django.utils.decorators import decorator_from_middleware_with_args
+from django.utils.decorators import (
+    decorator_from_middleware_with_args,
+    sync_and_async_middleware,
+)
 
 
+@sync_and_async_middleware
```

Comment:
I don't think this makes sense for a decorator. (It's used when loading middleware to determine if we need to adapt to the wrapped handler. That's not needed here.)

Reply:
I've removed this, although I'm not sure I completely follow the logic. `cache_page` is a decorator for a view, which runs `decorator_from_middleware_with_args(CacheMiddleware)(...)`. Does that not make it middleware in its own right? If so, shouldn't we show that it accepts both async and sync requests so that Django can "[optimize the middleware call stack to have as few sync/async transitions as possible](https://docs.djangoproject.com/en/4.2/topics/http/middleware/#asynchronous-support)"? 🙂

Reply:
Ah, I see. Thanks for explaining that with examples, it helps to follow the logic directly in the code 😄